crates/sql-sink/src/main.rs

```diff
@@ -27,49 +29,73 @@ const BACKOFF_MAX: Duration = Duration::from_secs(3600 * 24);
 
 #[connector(sink)]
 async fn start(config: SqlConfig, mut stream: impl ConsumerStream) -> Result<()> {
-    let mut sink = start_sink(config).await?;
-    while let Some(item) = stream.next().await {
-        let operation: Operation = serde_json::from_slice(item?.as_ref())?;
-        trace!(?operation);
-        sink.send(operation).await?;
+    let mut backoff = backoff_init()?;
+    let mut sink = start_sink(&mut backoff, &config).await?;
+
+    info!("Starting to process records");
+
+    while let Some(item_result) = stream.next().await {
+        match item_result {
+            Ok(item) => match serde_json::from_slice::<Operation>(item.as_ref()) {
+                Ok(operation) => {
+                    trace!(?operation);
+                    while let Err(e) = sink.send(operation.clone()).await {
```

Comment:
Lets move out of here into separate function. Backoff should be initialized at each attempt not at beginning of this loop

Reply:
oh, I forgot to reset the backoff after each record. I added a .reset after sink.send.

about move to a function, do you mean move the sink.send part?

Reply:
move entire to separate function or other abstraction so we can reason failure
```
while let Err(e) = sink.send(operation.clone()).await {
```

Reply:
Sure, Added all these logic in process_item function